eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -370,6 +431,595 @@ def test_update_user_in_tenant_user_not_found(self, param: str, value: str) -> N
         )
 
 
+@ddt.ddt
+class TestEnrollmentAPIIntegration(BaseAPIIntegrationTest, UsersAPIRequestMixin, EnrollmentAPIRequestMixin):
+    """Integration test suite for the Enrollment API"""
+
+    def setUp(self) -> None:
+        """Set up the test suite"""
+        self.course_id = settings["COURSE_ID"]
+        self.mode = "audit"
+        super().setUp()
+
+    @ddt.data("email", "username")
+    def test_create_enrollment_success(self, param: str) -> None:
+        """
+        Test creating a enrollment with a valid user, course and mode in a tenant.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+        - `create_enrollment`
+        - `check_edxapp_account_conflicts`
+        - `check_edxapp_enrollment_is_valid`
+
+        Expected result:
+        - The status code is 200.
+        - The enrollment is created successfully in the tenant with the provided data.
+        """
+        user_data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, user_data)
+        enrollment_data = {
+            param: user_data[param],
+            "course_id": self.course_id,
+            "mode": self.mode,
+        }
+
+        response = self.create_enrollment(self.tenant_x, enrollment_data)
+
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response_data["username"], user_data["username"])
+        self.assertEqual(response_data["mode"], enrollment_data["mode"])
+        self.assertEqual(response_data["course_id"], enrollment_data["course_id"])
+        self.assertTrue(response_data["is_active"])
+        self.assertIn("created", response_data)
+
+    @ddt.data(
+        ("mode", {"mode": ["This field is required."]}),
+        ("course_id", {"non_field_errors": ["You have to provide a course_id or bundle_id"]}),
+        ("username", {"non_field_errors": ["Email or username needed"]}),
+    )
```

Comment:
Should we also consider when neither the username nor the email are passed?

Reply:
Yes, the test is doing it. When we use ` enrollment_data.pop(param)`

Reply:
It considers only the username but not when both are missing. Am I correct?

Reply:
At least one (username or email) must be sent for the request to work. In this case, I did not include the email since we want to test when neither of these two fields is sent.

Reply:
This is the test case I'm referring to: 

```
curl -X 'POST' \
  'https://redwood.../eox-core/api/v1/enrollment/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFToken: ...' \
  -d '{
  "force": false,
  "is_active": true,
  "mode": "audit",
  "enrollment_attributes": [],
  "course_id": "course-v1:OpenedX+DemoX+DemoCourse",
  "created": "2021-10-11T10:06:17.876505-05:00"
}'
```
Can we include it as well?